projects/fluid_simulation/shaders.h

```diff
@@ -108,7 +104,7 @@ class Texture
 // This must match the CSInputs structure in assets/fluid_simulation/shaders/config.hlsli.
 struct alignas(16) ScalarInput
 {
-    ScalarInput(Texture* output)
+    ScalarInput(SimulationGrid* output)
```

Comment:
nit: should be `pOutput`

Reply:
Done.  I left some un modified because they are disappearing in the next PR.